feat: Add container mode support to bypass host executable validation

artemry-nv opened the pull request:
When MTT_CONTAINER_MODE environment variable is set, MTT will assume that test executables exist in the container even if they are not found on the host system. This allows running container-only tests without requiring the test binaries to be present on the host.

MTT test report for OSU test suite (test failures are expected): http://hpcweb.lab.mtl.com/hpc/mtr_scrap/users/artemry/scratch/ucx-enroot/20251126_173026_3676487_8006_funk10/

Bare-metal MTT run on funk:
http://hpcweb.lab.mtl.com/hpc/mtr_scrap/users/artemry/scratch/ucx_ompi/20251107_233618_361308_5704_funk08/

artemry-nv committed:
feat: Add container mode support to bypass host executable validation

When MTT_CONTAINER_MODE environment variable is set, MTT will assume
that test executables exist in the container even if they are not
found on the host system. This allows running container-only tests
without requiring the test binaries to be present on the host.

Bohatchuk reviewed (approved):
LGTM.
But please let's wait for additional approvals.

artemry-nv commented:
> LGTM. But please let's wait for additional approvals.

Thanks, the changes shouldn't impact non-container bare-metal mode.

alinask commented:
> > LGTM. But please let's wait for additional approvals.
> 
> Thanks, the changes shouldn't impact non-container bare-metal mode.

Hi @artemry-nv, can you please attach an mtt report for a job to make sure? one that ran with this code

artemry-nv commented:
> > > LGTM. But please let's wait for additional approvals.
> > 
> > 
> > Thanks, the changes shouldn't impact non-container bare-metal mode.
> 
> Hi @artemry-nv, can you please attach an mtt report for a job to make sure? one that ran with this code

Sure, added into the description.

Bohatchuk commented:
> Sure, added into the description.

Please correct me if I'm wrong, but the attached report is for the MTT job, in which MTT_CONTAINER_MODE was set.
Have you tried to run some regular MTT configuration but with your changes in mtt-legacy code?

alinask commented:
> > Sure, added into the description.
> 
> Please correct me if I'm wrong, but the attached report is for the MTT job, in which MTT_CONTAINER_MODE was set. Have you tried to run some regular MTT configuration but with your changes in mtt-legacy code?

I agree, @artemry-nv could you please check this as well?

artemry-nv commented:
> > Sure, added into the description.
> 
> Please correct me if I'm wrong, but the attached report is for the MTT job, in which MTT_CONTAINER_MODE was set. Have you tried to run some regular MTT configuration but with your changes in mtt-legacy code?

Yes, added bare-metal test report into the description.

alinask reviewed (approved):
Thanks Artem!
